db/tests/functions/operations/test_filter.py

```diff
@@ -0,0 +1,132 @@
+import re
+import pytest
+from datetime import datetime
+
+from db.utils import execute_query
+
+from db.functions.base import ColumnReference, Not, Literal, Empty, Equal, Greater
+from db.functions.operations.apply import apply_db_function_as_filter
+
+
+def _like(x, v):
+    return re.match(v.replace("%", ".*"), x) is not None
+
+
+def _ilike(x, v):
+    return re.match(v.replace("%", ".*").lower(), x.lower()) is not None
+
+
+database_functions = {
+    "is_null": lambda x: Empty([ColumnReference([x])]),
+    "is_not_null": lambda x: Not([Empty([ColumnReference([x])])]),
+    "eq": lambda x, v: Equal([ColumnReference([x]), Literal([v])]),
+    "gt": lambda x, v: Greater([ColumnReference([x]), Literal([v])]),
+}
+
+
+op_to_python_func = {
+    "is_null": lambda x, _: x is None,
+    "is_not_null": lambda x, _: x is not None,
+    "eq": lambda x, v: x == v,
+    "ne": lambda x, v: x != v,
+    "gt": lambda x, v: x > v,
+    "lt": lambda x, v: x < v,
+    "ge": lambda x, v: x >= v,
+    "le": lambda x, v: x <= v,
+    "like": _like,
+    "ilike": _ilike,
+    "not_ilike": lambda x, v: not _ilike(x, v),
+    "in": lambda x, v: x in v,
+    "not_in": lambda x, v: x not in v,
+    "any": lambda x, v: v in x,
+    "not_any": lambda x, v: v not in x,
+    "and": lambda x: all(x),
+    "or": lambda x: any(x),
+    "not": lambda x: not x[0]
+}
+
+
+ops_test_list = [
+    # is_null
+    ("varchar", "is_null", None, 5),
+    ("numeric", "is_null", None, 5),
+    ("date", "is_null", None, 5),
+    ("array", "is_null", None, 5),
+    # is_not_null
+    ("varchar", "is_not_null", None, 100),
+    ("numeric", "is_not_null", None, 100),
+    ("date", "is_not_null", None, 100),
+    ("array", "is_not_null", None, 100),
+    # eq
+    ("varchar", "eq", "string42", 1),
+    ("numeric", "eq", 1, 1),
+    ("date", "eq", "2000-01-01", 1),
+    ("array", "eq", "{0,0}", 1),
+    # ne
+    # ("varchar", "ne", "string42", 99),
```

Comment:
Please remove commented out code.

Reply:
Applies below as well.

Reply:
Weird, I remember you making this comment, then I wrote a reply, which is absent and then I didn't see this "conversation" anymore, so I presumed that you marked it as resolved.

Anyway, I originally replied that the commented tests serve to remind me of test cases that will likely be later implemented, but I'll just remove them to make review faster.